circle.yml

```diff
@@ -44,43 +48,40 @@ jobs:
 
 # ------------------------------------------------------------------------------
   release:
-    branch:
-      only:
-      - master
     docker:
       - image: mbgl/android-ndk-r21e:latest
     working_directory: ~/code
     steps:
       - checkout
-      - run:
-          name: Generate Bintray credentials
-          shell: /bin/bash -euo pipefail
-          command: |
-            if [ -n "${BINTRAY_USER}" ]; then
-              echo "BINTRAY_USER=$BINTRAY_USER
-              BINTRAY_API_KEY=$BINTRAY_API_KEY
-              GPG_PASSPHRASE=$GPG_PASSPHRASE"
-            fi
+      - init-aws
       - run:
           name: Update version name
           command: |
             if [[ $CIRCLE_TAG == v* ]]; then
               sed -i -e "s/^VERSION_NAME=.*/VERSION_NAME=${CIRCLE_TAG:1}/" library/gradle.properties
-            fi      
+            elif [[ $CIRCLE_BRANCH == master ]]; then
+              COMMIT_SHA=$(git rev-parse --short HEAD)
+              sed -i -e "s/-SNAPSHOT.*/-${COMMIT_SHA}-SNAPSHOT/" library/gradle.properties
+            fi
       - run:
           name: Build libraries
           command: make release
       - deploy:
-          name: Publish to Bintray
+          name: Publish to the SDK registry
           command: |
             if [[ $CIRCLE_BRANCH == master ]] || [[ $CIRCLE_TAG == v* ]]; then
-              version=$(cat library/gradle.properties | grep "VERSION_NAME")
-              if [[ $version != *"SNAPSHOT"* ]]; then
-                make publish-stable
-              else
-                make publish-snapshot-to-artifactory
-              fi
+              make publish
```

Comment:
we should create use `./gradlew library:mapboxSDKRegistryPublish` when releasing tag was used. And only skip the `mapboxSDKRegistryPublish` when the snapshot is uploaded.

Reply:
Added!